saber/dvid_access/dvid_access.py

```diff
@@ -0,0 +1,188 @@
+            break
+        except HTTPError as e: 
+            if attempts < 3:
+                attempts += 1
+                print("These are the dimensions: ")
+                print(data.shape)
+                print("This is the data type:")
+                print(data.dtype)
+                print("Specified data type was:")
+                print(args.dtype)
+                print("Specified image type")
+                print(args.itype)
+                print("Obtained HTTP error from server. Trial {}".format(attempts))
+                print("The error: {}".format(e))
+            else:
+                raise Exception("Failed 3 times: {}".format(e))
+    # Clean up
+
+def main():
+    parser = argparse.ArgumentParser(description='dvid processing script')
+    parent_parser = argparse.ArgumentParser(add_help=False)
+    subparsers = parser.add_subparsers(title='commands')
+
+    parser.set_defaults(func=lambda _: parser.print_help())
+
+    parent_parser.add_argument('-b', '--bucket', default=None, help='S3 bucket to save to or load from')
+
+    parent_parser.add_argument('--uuid', required=False, default=None, help='Root UUID of the repository')
+    parent_parser.add_argument('--alias', required=False, default='', help='Readable UUID Tag')
+    parent_parser.add_argument('--data_instance', required=True, help='Name of data instance within repository ')
+    parent_parser.add_argument('--datatype', required=False, help='data type of the instance (uint8, uint16, uint64) defaults to uint8')
+    parent_parser.add_argument('--type', required=False, help='type of the resource (uint8blk, labelblk) defaults to uint8blk')
+    parent_parser.add_argument('--host', required=True, help='Name of DVID host')
+
+    parent_parser.add_argument('--res', type=int, help='Resolution')
+    parent_parser.add_argument('--xmin', type=int, default=0, help='Xmin')
+    parent_parser.add_argument('--xmax', type=int, default=1, help='Xmax')
+    parent_parser.add_argument('--ymin', type=int, default=0, help='Ymin')
+    parent_parser.add_argument('--ymax', type=int, default=1, help='Ymax')
+    parent_parser.add_argument('--zmin', type=int, default=0, help='Zmin')
+    parent_parser.add_argument('--zmax', type=int, default=1, help='Zmax')
+
+    push_parser = subparsers.add_parser(
+        'push', help='Push images to dvid from input file or s3 bucket', parents=[parent_parser]
+    )
+    pull_parser = subparsers.add_parser(
+        'pull', help='Pull images from dvid and optionally save to AWS S3',
+        parents=[parent_parser]
+    )
+
+    push_parser.add_argument('-i', '--input', required=True, help='Input file')
+    push_parser.add_argument('--source', required=False, help='Source channel for upload')
+    push_parser.set_defaults(func=DVID_push_cutout)
+
+    pull_parser.add_argument('-o', '--output', required=True, help='Output file')
+    pull_parser.add_argument(
```

Comment:
how does this work with the s3 wrap functionality? should this s3 functionality be subsumed by s3wrap?